aikit/models/ordinal.py

```diff
@@ -0,0 +1,364 @@
+        else:
+            return probas
+        
+        
+    def predict(self, X):
+        probas = self.predict_proba(X)
+        
+        classes = self.classes_
+        
+        if self._mono_target:
+            y_hat = classes[probas.argmax(axis=1)]
+        
+        else:
+            
+            all_pred = [ c[p.argmax(axis=1)][:, np.newaxis] for p, c in zip(probas, classes)]
+            y_hat = np.concatenate(all_pred, axis=1)
+            
+        return y_hat.astype(self._target_dtype)
+
+
+
+class RegressorFromClassifier(BaseEstimator, RegressorMixin):
+    """ Transform a Classifier into a regressor
+    
+    does it by clustering the target and fit a classification
+    """
+    
+    def __init__(self,
+                 classifier,
+                 strategy="kmeans",
+                 n_bins=10,
+                 y_clusterer=None
+                 ):
+        self.classifier=classifier
+        self.strategy=strategy
+        self.n_bins=n_bins
+        self.y_clusterer=y_clusterer
+        
+        
+    def get_default_y_cluster(self, y=None):
+        return KBinsDiscretizer(n_bins=self.n_bins, strategy=self.strategy, encode="ordinal")
+  
+    
+    def fit(self, X, y):
+        
+        self._mono_target = y.ndim == 1
+        
+        if self.y_clusterer is None:
+            y_clusterer = self.get_default_y_cluster(y)
+        else:
+            y_clusterer = self.y_clusterer
+            # TODO : check that it is a clusterer
+            
+        if not is_classifier(self.classifier):
+            raise TypeError("classifier should be a classifer")
+            
+        yd2 = make2dimensions(y)
+        
+        if hasattr(y_clusterer, "fit_predict"):
+            y_cl = y_clusterer.fit_predict(yd2)
+        else:
+            y_cl = y_clusterer.fit_transform(yd2).astype('int32')
+        
+        if y_cl.ndim == 1:
+            y_cl = y_cl[:, np.newaxis]
+
+        if self._mono_target and y_cl.shape[1] > 1:
+            raise ValueError("The cluster should return only 1 dimensional clusters")
+ 
+        self._mono_cluster = y_cl.shape[1] == 1
+#        if self._mono_cluster != self._mono_target:
+#            raise NotImplementedError("TODO")
+#        
+        self.classifier.fit(X, y_cl) # fit classifier on result of cluster
+        
+        if self._mono_cluster:
+            classes = [self.classifier.classes_]
+        else:
+            classes = self.classifier.classes_
+            
+        all_mean_mapping = self.compute_y_mean(yd2, y_cl)
+        
+        all_y_mean_mapping_matrix = []
+        for classe, y_mean_mapping in zip(classes, all_mean_mapping):
+            mat = np.concatenate([y_mean_mapping[cl] for cl in classe], axis=0)
+            all_y_mean_mapping_matrix.append(mat)
+            
+        self._all_y_mean_matrix = all_y_mean_mapping_matrix
+        
+        return self
+    
+    
+    def compute_y_mean(self, yd2, y_cl):
+        assert y_cl.ndim == 2
```

Comment:
same here: maybe add some doc to describe method intent

Reply:
done